### PR TITLE
Verify repeatable hl7 fields

### PR DESCRIPTION
**Pull request by jimmyfagan**
This PR includes several test updates to make sure existing tests include coverage for repeating elements. It also includes some mapping updates to correctly handle any repeatable elements that were previously insufficient.

Test Steps:
1. Run the integration tests

## Changes
- Mapping changes, mostly to change the level at which `specs` is used to allow for repetitions to be handled, and to include an element's index in hl7 path variables
- Some FHIR->HL7 mappings adjusted to check for the standard mapped HL7 field along with repetitions tracked via extensions for cases where the mapping from HL7->FHIR does not preserve cardinality. 

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] Added tests?

## Linked Issues
- Addresses #12901

### CODE REVIEW

File: prime-router/metadata/fhir_mapping/hl7/datatype/Annotation.yml [hunk 1]
```diff
@@ -54,7 +54,20 @@ extension_1-noteSource:
       expressionType: HL7Spec
       type: STRING
 
-extension_2-noteType:
+extension_2-noteComment:
```
Comment (brick-green) [inline prime-router/metadata/fhir_mapping/hl7/datatype/Annotation.yml:57]: Is there a mapping for NTE to Annotation? I don't see it in the inventory

**Review by brick-green — APPROVED** [review]
Just some questions. Looks good to me :)

File: prime-router/metadata/fhir_mapping/hl7/segments/ORC/OBRExtension.yml [hunk 2]
```diff
@@ -36,14 +36,15 @@ extension:
       vars:
         obr10: STRING_ALL, OBR.10
       condition: $obr10 NOT_NULL
+      specs: OBR.10
```
[the remark this answers is not in the document]
Comment (jimmyfagan) [reply prime-router/metadata/fhir_mapping/hl7/segments/ORC/OBRExtension.yml:39]: Apparently not, though I'm not sure why not 🤷 Might be good to add them for the repeated fields though as a convention, I just forgot about that this time around. I can try to go through these and do that before merging.

File: prime-router/metadata/fhir_mapping/hl7/datatype/Annotation.yml (hunk 1 above)
Comment (jimmyfagan) [reply prime-router/metadata/fhir_mapping/hl7/datatype/Annotation.yml:57]: Not really, it seems to be based in [NTE[ServiceRequest]](https://docs.google.com/spreadsheets/d/1ycH9LSI0xRyZqiRmh233wdpmJ-MyjzPFe4015X2TDDY/edit#gid=0), that could probably be revisited, but for the purposes of this story, just decided to try to fit into the existing files.

Comment (mkalish) [reply prime-router/metadata/fhir_mapping/hl7/datatype/Annotation.yml:57]: Yea, there is no official mapping for some reason, so this is just what made sense at the time

**Review by mkalish — APPROVED** [review]
Great stuff!  Appreciate all the work that went in to.  Changes look good to me, the only comment I have is maybe adding a few new HL7 files (or updating ones) to capture the repeat behavior added here.

**Pull request merged**